### ISSUE TITLE
How to solve the image resize shift problem ？

### ISSUE THREAD

**Issue by fat-921** [open-mmlab/mmsegmentation#3379]
When I used the original image size for training, but reduced it to double size for verification and testing, the mask map position offset problem occurred. I modified the img_scale, trained three models separately, just like the picture below:
**my original image size is 480x320**
![image](https://github.com/open-mmlab/mmsegmentation/assets/66895706/d93e2490-041d-46dc-acef-97925b7afcc0)
![image](https://github.com/open-mmlab/mmsegmentation/assets/66895706/f430f8b9-f243-4f62-a49d-a839db989f30)
![image](https://github.com/open-mmlab/mmsegmentation/assets/66895706/66498a6f-958f-405c-baa7-61564fc20c5d)

Here is my training config file: `knet-s3_r50-d8_pspnet_8xb2-adamw-zd.py`
```
# custom config
data_root = '/mnt/share/datasets/zhidai_0306/'
log_file_name = './work_dirs/knet-s3_r50-d8_pspnet_8xb2-adamw-zd'
batch_size = 8
max_iter = 100
CLASSES = ['dj', 'djxc', 'djqs', 'djyx', 'bt', 'btxc', 'btqs', 'cpmc']
classes_num = 9
PALETTE = [[0, 0, 0], [128, 0, 0], [0, 128, 0], [128, 128, 0], [0, 0, 128],
           [128, 0, 128], [0, 128, 128], [128, 128, 128]]
class_weight = None
METAINFO = {'classes':CLASSES, 'palette':PALETTE}

img_scale = (120, 80)   # (480, 320)  (240, 160), (120, 80)


data_preprocessor = dict(
    type='SegDataPreProcessor',
    bgr_to_rgb=True,
    mean=[29.828, 28.19, 99.768],
    std=[27.69, 29.3, 94.137],
    pad_val=0,
    seg_pad_val=255,
    size=(256, 256))

default_hooks = dict(
    checkpoint=dict(by_epoch=True, interval=10, type='CheckpointHook', save_best='auto', max_keep_ckpts=3),
    logger=dict(interval=10, log_metric_by_epoch=False, type='LoggerHook'),
    param_scheduler=dict(type='ParamSchedulerHook'),
    sampler_seed=dict(type='DistSamplerSeedHook'),
    timer=dict(type='IterTimerHook'),
    visualization=dict(type='SegVisualizationHook'))
default_scope = 'mmseg'
env_cfg = dict(
    cudnn_benchmark=True,
    dist_cfg=dict(backend='nccl'),
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0))
gpu_ids = [1]
img_ratios = [0.5, 0.75, 1.0, 1.25, 1.5, 1.75]
launcher = 'none'
load_from = None
log_level = 'INFO'
log_processor = dict(by_epoch=True)
model = dict(
    type='EncoderDecoder',
    data_preprocessor=data_preprocessor,
    auxiliary_head=dict(
        align_corners=False,
        channels=256,
        concat_input=False,
        dropout_ratio=0.1,
        in_channels=1024,
        in_index=2,
        loss_decode=dict(
            loss_weight=0.4, type='CrossEntropyLoss', use_sigmoid=False),
        norm_cfg=dict(requires_grad=True, type='SyncBN'),
        num_classes=classes_num,
        num_convs=1,
        type='FCNHead'),
    backbone=dict(
        contract_dilation=True,
        depth=50,
        dilations=(1, 1, 2, 4),
        norm_cfg=dict(requires_grad=True, type='SyncBN'),
        norm_eval=False,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        strides=(1, 2, 1, 1),
        style='pytorch',
        type='ResNetV1c'),
    decode_head=dict(
        kernel_generate_head=dict(
            align_corners=False,
            channels=512,
            dropout_ratio=0.1,
            in_channels=2048,
            in_index=3,
            loss_decode=dict(
                loss_weight=1.0, type='CrossEntropyLoss', use_sigmoid=False),
            norm_cfg=dict(requires_grad=True, type='SyncBN'),
            num_classes=classes_num,
            pool_scales=(1, 2, 3, 6),
            type='PSPHead'),
        kernel_update_head=[
            dict(
                conv_kernel_size=1,
                dropout=0.0,
                feat_transform_cfg=dict(
                    act_cfg=None, conv_cfg=dict(type='Conv2d')),
                feedforward_channels=2048,
                ffn_act_cfg=dict(inplace=True, type='ReLU'),
                in_channels=512,
                kernel_updator_cfg=dict(
                    act_cfg=dict(inplace=True, type='ReLU'),
                    feat_channels=256,
                    in_channels=256,
                    norm_cfg=dict(type='LN'),
                    out_channels=256,
                    type='KernelUpdator'),
                num_classes=classes_num,
                num_ffn_fcs=2,
                num_heads=8,
                num_mask_fcs=1,
                out_channels=512,
                type='KernelUpdateHead',
                with_ffn=True),
            dict(
                conv_kernel_size=1,
                dropout=0.0,
                feat_transform_cfg=dict(
                    act_cfg=None, conv_cfg=dict(type='Conv2d')),
                feedforward_channels=2048,
                ffn_act_cfg=dict(inplace=True, type='ReLU'),
                in_channels=512,
                kernel_updator_cfg=dict(
                    act_cfg=dict(inplace=True, type='ReLU'),
                    feat_channels=256,
                    in_channels=256,
                    norm_cfg=dict(type='LN'),
                    out_channels=256,
                    type='KernelUpdator'),
                num_classes=classes_num,
                num_ffn_fcs=2,
                num_heads=8,
                num_mask_fcs=1,
                out_channels=512,
                type='KernelUpdateHead',
                with_ffn=True),
            dict(
                conv_kernel_size=1,
                dropout=0.0,
                feat_transform_cfg=dict(
                    act_cfg=None, conv_cfg=dict(type='Conv2d')),
                feedforward_channels=2048,
                ffn_act_cfg=dict(inplace=True, type='ReLU'),
                in_channels=512,
                kernel_updator_cfg=dict(
                    act_cfg=dict(inplace=True, type='ReLU'),
                    feat_channels=256,
                    in_channels=256,
                    norm_cfg=dict(type='LN'),
                    out_channels=256,
                    type='KernelUpdator'),
                num_classes=classes_num,
                num_ffn_fcs=2,
                num_heads=8,
                num_mask_fcs=1,
                out_channels=512,
                type='KernelUpdateHead',
                with_ffn=True),
        ],
        num_stages=3,
        type='IterativeDecodeHead'),
    pretrained='open-mmlab://resnet50_v1c',
    test_cfg=dict(mode='whole'),
    train_cfg=dict())

optim_wrapper = dict(
    clip_grad=dict(max_norm=1, norm_type=2),
    optimizer=dict(lr=0.0001, type='AdamW', weight_decay=0.0005),
    type='OptimWrapper')
optimizer = dict(lr=0.01, momentum=0.9, type='SGD', weight_decay=0.0005)
param_scheduler = [
    dict(
        begin=0, by_epoch=True, end=10, start_factor=0.001,
        type='LinearLR'),
    dict(
        begin=10,
        by_epoch=True,
        end=100,
        milestones=[
            70,
            80,
        ],
        type='MultiStepLR'),
]
resume = False


train_cfg = dict(type='EpochBasedTrainLoop', max_epochs=max_iter, val_interval=10)
train_dataloader = dict(
    batch_size=batch_size,
    dataset=dict(
        type='PascalVOCDataset',
        metainfo=METAINFO,
        data_root=data_root,
        img_suffix='.bmp',
        seg_map_suffix='.png',
        ann_file='labelme2voc/train.txt',
        data_prefix=dict(
            img_path='', seg_map_path='labelme2voc/SegmentationClassPNG'),
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(reduce_zero_label=False, type='LoadAnnotations'),
            dict(
                keep_ratio=True,
                ratio_range=(0.5, 2.0),
                scale=(480, 320),
                type='RandomResize'),
            dict(
                cat_max_ratio=0.75, crop_size=(256, 256), type='RandomCrop'),
            dict(prob=0.5, type='RandomFlip'),
            dict(type='PhotoMetricDistortion'),
            dict(type='PackSegInputs'),
        ]),
    num_workers=2,
    persistent_workers=True,
    sampler=dict(shuffle=True, type='DefaultSampler'))

val_cfg = dict(type='ValLoop')
val_dataloader = dict(
    batch_size=1,
    dataset=dict(
        type='PascalVOCDataset',
        metainfo=METAINFO,
        data_root=data_root,
        img_suffix='.bmp',
        seg_map_suffix='.png',
        ann_file='labelme2voc/val.txt',
        data_prefix=dict(img_path='', seg_map_path='labelme2voc/SegmentationClassPNG'),
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(keep_ratio=True, scale=img_scale, type='Resize'),
            dict(reduce_zero_label=False, type='LoadAnnotations'),
            dict(type='PackSegInputs'),
        ]),
    num_workers=4,
    persistent_workers=True,
    sampler=dict(shuffle=False, type='DefaultSampler'))
val_evaluator = dict(
    iou_metrics=[
        'mIoU',
    ], type='IoUMetric')


test_cfg = dict(type='TestLoop')
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(keep_ratio=True, scale=img_scale, type='Resize'),
    dict(reduce_zero_label=False, type='LoadAnnotations'),
    dict(type='PackSegInputs'),
]
test_dataloader = dict(
    batch_size=1,
    dataset=dict(
        type='PascalVOCDataset',
        metainfo=METAINFO,
        data_root=data_root,
        img_suffix='.bmp',
        seg_map_suffix='.png',
        ann_file='labelme2voc/val.txt',
        data_prefix=dict(
            img_path='',
            seg_map_path='labelme2voc/SegmentationClassPNG'),
        pipeline=test_pipeline),
    num_workers=4,
    persistent_workers=True,
    sampler=dict(shuffle=False, type='DefaultSampler'))
test_evaluator = dict(
    iou_metrics=[
        'mIoU',
    ], type='IoUMetric')
tta_model = dict(type='SegTTAModel')
tta_pipeline = [
    dict(backend_args=None, type='LoadImageFromFile'),
    dict(
        transforms=[
            [
                dict(keep_ratio=True, scale_factor=0.5, type='Resize'),
                dict(keep_ratio=True, scale_factor=0.75, type='Resize'),
                dict(keep_ratio=True, scale_factor=1.0, type='Resize'),
                dict(keep_ratio=True, scale_factor=1.25, type='Resize'),
                dict(keep_ratio=True, scale_factor=1.5, type='Resize'),
                dict(keep_ratio=True, scale_factor=1.75, type='Resize'),
            ],
            [
                dict(direction='horizontal', prob=0.0, type='RandomFlip'),
                dict(direction='horizontal', prob=1.0, type='RandomFlip'),
            ],
            [
                dict(type='LoadAnnotations'),
            ],
            [
                dict(type='PackSegInputs'),
            ],
        ],
        type='TestTimeAug'),
]

vis_backends = [
    dict(type='LocalVisBackend'),
]
visualizer = dict(
    name='visualizer',
    type='SegLocalVisualizer',
    vis_backends=[
        dict(type='LocalVisBackend'),
    ])
work_dir = log_file_name
```
